platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationLogger.java

```diff
@@ -67,7 +67,10 @@ protected Integer initialValue() {
 
     private static final LoggingDeprecatedFeatureHandler DEPRECATED_FEATURE_HANDLER = new LoggingDeprecatedFeatureHandler();
 
+    private static boolean initialized = false;
+
     public synchronized static void init(WarningMode warningMode, BuildOperationProgressEventEmitter buildOperationProgressEventEmitter, Problems problemsService, ProblemStream problemStream) {
+        initialized = true;
```

Comment:
I would have expected to track/check this state in LoggingDeprecatedFeatureHandler, but otherwise this seems like the right way to go for me

Reply:
The reasoning being that `DeprecationLogger` is almost completely stateless, except for the `ThreadLocal` `DISABLE_COUNT`. The vast majority of mutable state related to deprecations is encapsulated in `LoggingDeprecatedFeatureHandler`. 

It seems more of a special case rather than the normal case that `DISABLE_COUNT` lives in `DeprecationLogger`, and I would argue we should also move that into `LoggingDeprecatedFeatureHandler` in order to keep all our state in one place